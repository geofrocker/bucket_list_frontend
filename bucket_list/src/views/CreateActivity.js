import React, {Component} from 'react';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';
import {Redirect} from 'react-router-dom';
import axios from 'axios'

class CreateActivity extends Component{
    constructor(props){
        super(props);
        this.state = {description:'', token: '',isAuthorized:false};
        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
        axios({
            // url: 'https://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/',
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/',
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                this.setState({isAuthorized:true});
            })
            .catch((xhr) =>{
                alert(xhr);
            });
    }

    handleDescriptionChange(event){
        this.setState({description: event.target.value})
    }

    handleSubmit(event){
        event.preventDefault();
        let data = {
            description: this.state.description,
        };

        axios({
            // url: 'http://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/',
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/',
            method: 'POST',
            datatype: "json",
            data: data,
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }

        })
            .then((response)=>{
                this.setState({redirect:true});
            })
            .catch((xhr) =>{
                console.log(JSON.stringify(xhr));
                alert(xhr.response.data.error);
            });
    }

    render(){
        if (this.state.isAuthorized === false){
            return(
                <div>
                    <article className="content item-editor-page">
                        <div className="card card-block">
                            <p>Unauthorized! Please log in</p>
                        </div>
                    </article>
                </div>
            )
        }
        else if (this.state.redirect === true){
            return <Redirect to="/bucketlists/view"/>
        }
        return(
            <div>
                <Header/>
                <Sidebar/>
                <article className="content item-editor-page">
                    <div className="title-block">
                        <h3 className="title"> Add new Activity
                            <span className="sparkline bar"></span>
                        </h3>
                    </div>

                    <form name="item" method="post" onSubmit={this.handleSubmit()}>
                        <div className="card card-block">
                            <div className="form-group row">
                                <label className="col-sm-2 form-control-label text-xs-right"
                                       onChange={this.handleUpdateDescription}>Description</label>
                                <div className="col-sm-10">
                                    <div className="wyswyg">
                                        <div className="editor" id="div_description"></div>
                                    </div>
                                </div>
                            </div>

                            <div className="col-sm-10 col-sm-offset-2">
                                <button type="submit" className="btn btn-primary">
                                    Update
                                </button>
                            </div>

                        </div>
                    </form>
                </article>
            </div>

        )
    }
}
export default CreateActivity;