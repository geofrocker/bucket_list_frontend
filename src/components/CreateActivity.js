import React, {Component} from 'react';
import Header from './Header';
import Sidebar from './Sidebar';
import {Redirect} from 'react-router-dom';
import axios from 'axios'
import swal from 'sweetalert';

class CreateActivity extends Component{
    /**
     * This component is used to create activities or items.
     * It handles the following methods:
     *  handleDescriptionChange --> used to set the description state in the constructor
     *  handleSubmit --> Used to post data in the end point
     *  render --> Used to return jsx
     */

    constructor(props){
        super(props);
        let bucket_id = this.props.match.params.bucket_id;
        this.state = {description:'', token: '',isAuthorized:false, bucket_id:bucket_id, redirect: false, login_redirect:false};
        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
        axios({
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/' + this.state.bucket_id,
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then(()=>{
                this.setState({isAuthorized:true});
            })
            .catch(() =>{
                this.setState({redirect:true})
            });
    }

    handleDescriptionChange(event){
        this.setState({description: event.target.value})
    }

    handleSubmit(event){
        event.preventDefault();
        let data = {
            description: this.state.description,
        };

        axios({
            url: 'http://127.0.0.1:5000/api/v1/bucketlists/'+ this.state.bucket_id + '/items',
            method: 'POST',
            datatype: "json",
            data: data,
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }

        })
            .then(()=>{
                this.setState({redirect:true});
            })
            .catch((xhr) =>{
                swal("Error!", xhr.response.data.error, "error");
            });
    }

    render(){
        if(this.state.login_redirect){
            return(<Redirect to={'/login/'}/>)
        }

        if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">

                    </article>
                </div>
            )
        }

         if (this.state.redirect){
                    return <Redirect to={'/bucketlists/'+ this.state.bucket_id + '/items/view/'}/>

         }

         return(
                <div>
                    <Header/>
                    <Sidebar/>
                    <article className="content item-editor-page">
                        <div className="title-block">
                            <h3 className="title"> Add new Activity
                                <span className="sparkline bar"></span>
                            </h3>
                        </div>

                        <form method="post" id="create_activity_form" onSubmit={this.handleSubmit}>
                            <div className="card card-block">
                                <div className="form-group row">
                                    <label className="col-sm-2 form-control-label text-xs-right">
                                        Description:
                                    </label>

                                    <div className="col-sm-10">
                                        <input type="text" name="description" value={this.state.description} onChange={this.handleDescriptionChange} className="form-control boxed" placeholder="" />
                                    </div>
                                </div>


                                <div className="col-sm-10 col-sm-offset-2">
                                    <button type="submit" className="btn btn-primary">
                                        Add
                                    </button>
                                </div>

                            </div>
                        </form>
                    </article>
                </div>

            )
    }
}


export default CreateActivity;