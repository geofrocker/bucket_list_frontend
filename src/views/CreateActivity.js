import React, {Component} from 'react';
import Header from '../components/Header';
import Sidebar from '../components/Sidebar';
import {Redirect, Link} from 'react-router-dom';
import axios from 'axios'
import swal from 'sweetalert';

class CreateActivity extends Component{
    constructor(props){
        super(props);
        let bucket_id = this.props.match.params.bucket_id;
        this.state = {description:'', token: '',isAuthorized:false, bucket_id:bucket_id};
        this.handleDescriptionChange = this.handleDescriptionChange.bind(this);
        this.handleSubmit = this.handleSubmit.bind(this);
    }

    componentDidMount() {
        axios({
            url: 'https://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/' + this.state.bucket_id,
            method: "GET",
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }
        })
            .then((response)=>{
                this.setState({isAuthorized:true});
            })
            .catch((xhr) =>{
                console.log(JSON.stringify(xhr))
            });
    }

    handleDescriptionChange(event){
        this.setState({description: event.target.value})
    }

    handleSubmit(event){
        event.preventDefault();
        let data = {
            description: this.state.description,
        };

        axios({
            url: 'https://ridge-bucket-list-api.herokuapp.com/api/v1/bucketlists/'+ this.state.bucket_id + '/items',
            method: 'POST',
            datatype: "json",
            data: data,
            headers: {
                'token': window.localStorage.getItem('token'),
                'Content-Type': 'application/json'
            }

        })
            .then((response)=>{
                this.setState({redirect:true});
            })
            .catch((xhr) =>{
                console.log(JSON.stringify(xhr));
                swal("Error!", xhr.response.data.error, "error");
            });
    }

    render(){
        if (!this.state.isAuthorized){
            return(
                <div>
                    <article className="content item-editor-page">

                    </article>
                </div>
            )
        }

         if (this.state.redirect){
                    return <Redirect to={'/bucketlists/'+ this.state.bucket_id + '/items/view/'}/>

         }

         return(
                <div>
                    <Header/>
                    <Sidebar/>
                    <article className="content item-editor-page">
                        <div className="title-block">
                            <h3 className="title"> Add new Activity
                                <span className="sparkline bar"></span>
                            </h3>
                        </div>

                        <form method="post" id="create_activity_form" onSubmit={this.handleSubmit}>
                            <div className="card card-block">
                                <div className="form-group row">
                                    <label className="col-sm-2 form-control-label text-xs-right">
                                        Name:
                                    </label>

                                    <div className="col-sm-10">
                                        <input type="text" name="description" value={this.state.description} onChange={this.handleDescriptionChange} className="form-control boxed" placeholder="" />
                                    </div>
                                </div>


                                <div className="col-sm-10 col-sm-offset-2">
                                    <button type="submit" className="btn btn-primary">
                                        Add
                                    </button>
                                </div>

                            </div>
                        </form>
                    </article>
                </div>

            )
    }
}
export default CreateActivity;