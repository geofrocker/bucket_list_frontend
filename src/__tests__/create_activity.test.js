import React from 'react';
import CreateActivity from '../views/CreateActivity';

import {mount} from 'enzyme';
import toJson from 'enzyme-to-json';
import {MemoryRouter} from 'react-router-dom';

describe('<CreateActivity />', ()=>{
    beforeEach(() => {
        global.localStorage = {
            token: 'someToken',
            getItem: function () {
                return 'someToken'
            }
        };

    });

    it('contains create activity form', function () {
        const props = {
            match:{
                params: {bucket_id:1}
            }
        };
        const wrapper = mount(
            <MemoryRouter>
                <CreateActivity {...props}/>
            </MemoryRouter>);

        const preventDefault = jest.fn();
        wrapper.setState({description:"test description", isAuthorized:true});

        setTimeout(() => {
            expect(wrapper.find("#create_activity_form").length === 1);
            wrapper.find("#create_activity_form").simulate('submit', {preventDefault})

        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();


    });

    it('should change values in the form', function () {
        const props = {
            match:{
                params: {bucket_id:1}
            }
        };
        const wrapper = mount(
            <MemoryRouter>
                <CreateActivity {...props} />
            </MemoryRouter>);
        setTimeout(() => {
            wrapper.find("bucket_name").simulate('change', {target: {value: 'test bucket name'}});
            wrapper.find("description").simulate('change', {target: {value:'test description'}});
            wrapper.find("category").simulate('change', {target: {value:'test category'}});
        }, 60);
        expect(toJson(wrapper)).toMatchSnapshot();
    })



});

