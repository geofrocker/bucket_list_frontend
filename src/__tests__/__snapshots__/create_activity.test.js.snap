// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateActivity /> contains create activity form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "oveyyc",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "oveyyc",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CreateActivity
      match={
        Object {
          "params": Object {
            "bucket_id": 1,
          },
        }
      }
    >
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </CreateActivity>
  </Router>
</MemoryRouter>
`;

exports[`<CreateActivity /> should change values in the form 1`] = `
<MemoryRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "4f3i71",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "4f3i71",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <CreateActivity
      match={
        Object {
          "params": Object {
            "bucket_id": 1,
          },
        }
      }
    >
      <div>
        <article
          className="content item-editor-page"
        />
      </div>
    </CreateActivity>
  </Router>
</MemoryRouter>
`;
